### DISCUSSION TITLE
Prevent cookie from being displayed in the log

### DISCUSSION THREAD

**Discussion by f2face** [streamlink/streamlink#5338] [Q&A] [answered]
### Description

I use Streamlink with `--http-cookie` and `--loglevel=debug` to download videos from a website that requires cookie. Is there a Streamlink command to prevent the cookie from being displayed in the log?

So, instead of this:
```
[cli][debug]  --http-cookie=[('cookie_key', 'value_abcdef123456')]
```

I expect it to be something like this:
```
[cli][debug]  --http-cookie=[('cookie_key', '[REDACTED]')]
```

**Comment by bastimeyer** [comment, marked as answer]
No, there's currently no way to mask sensitive data of generic CLI arguments.

Plugin arguments on the other hand support that [because it's a custom API which Streamlink provides](https://streamlink.github.io/api.html#streamlink.plugin.pluginargument). The main CLI arguments however are defined via the `argparse` module of Python's standard library. If we'd add support for sensitive data, then we'd have to extend streamlink_cli's argparser with custom actions that receive additional data like a boolean sensitive flag that could then be checked when logging the current arguments.

You could open a feature request for that, but I don't know if it's worth implementing that just for the `--http-cookie` CLI argument.
